### PR TITLE
ArkType validator middleware

### PR DESCRIPTION
**Pull request by MonsterDeveloper**
A validator middleware using [ArkType](https://arktype.io). Built after the `zValidator`.

Example usage:

```ts
import { type } from 'arktype'
import { arktypeValidator } from '@hono/arktype-validator'

const schema = type({
  name: 'string',
  age: 'number'
})

app.post('/author', arktypeValidator('json', schema), (c) => {
  const data = c.req.valid('json')
  return c.json({
    success: true,
    message: `${data.name} is ${data.age}`,
  })
})
```

### CODE REVIEW

File: packages/arktype-validator/src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+  const route = app.post(
+    '/author',
+    arktypeValidator('json', jsonSchema),
+    arktypeValidator('query', querySchema),
+    (c) => {
+      const data = c.req.valid('json')
+      const query = c.req.valid('query')
+
+      return c.jsonT({
```
Comment (yusukebe) [inline packages/arktype-validator/src/index.test.ts:29]: `c.jsonT()` is now deprecated. Use `c.json()`.

**Comment by yusukebe** [comment]
Hi @MonsterDeveloper 

Cool! I didn't know ArkType, but it's simple and seems to be great. I've left one minor comment. Please check it out.

**Comment by yusukebe** [comment]
@MonsterDeveloper 

And two things:

1. Please add the CI to there: https://github.com/honojs/middleware/tree/main/.github/workflows
2. Add a build command below here: https://github.com/honojs/middleware/blob/616afbcd03928dbec40089cec638e15eea3e2030/package.json#L27

**Comment by MonsterDeveloper** [comment]
@yusukebe done!

**Comment by yusukebe** [comment]
@MonsterDeveloper 

Thank you! I'll check it again later and release it if everything is ok.

**Comment by yusukebe** [comment]
Hi @MonsterDeveloper 

I've tried this ArkType Validator Middleware again. Feels great! Let's land it!

**Pull request merged**